From f2192bcc61b918290fbf9754aacd9ca058ee2219 Mon Sep 17 00:00:00 2001
From: Pengpeng Cui <pengpengx.cui@intel.com>
Date: Mon, 14 Sep 2015 14:52:22 +0800
Subject: [PATCH 465/466] atomisp2: css2401 and
 2401_legacy-irci_master_20150911_0724

Merge CSS2.1 for css2401 and 2401_legacy
for irci_master_20150911_0724 firmware tag

Change-Id: I4e8c9d5e911857b91eff2d9a262ada268c70ea93
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-43419
Signed-off-by: Pengpeng Cui <pengpengx.cui@intel.com>
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/73944
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/css/css_api_version.h   |  6 +++++-
 .../media/pci/atomisp2/css/hive_isp_css_include/mpmath.h      | 11 +++++++++++
 .../camera/drivers/media/pci/atomisp2/css/ia_css.h            |  2 +-
 .../camera/drivers/media/pci/atomisp2/css/ia_css_host_data.h  |  2 +-
 .../drivers/media/pci/atomisp2/css/ia_css_pipe_public.h       |  3 ++-
 .../camera/drivers/media/pci/atomisp2/css/ia_css_types.h      |  2 +-
 .../drivers/media/pci/atomisp2/css/ia_css_version_data.h      |  2 +-
 .../media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr.host.c  |  2 +-
 .../media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr.host.h  |  2 +-
 .../media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr_param.h |  2 +-
 .../media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr_types.h |  2 +-
 .../camera/drivers/media/pci/atomisp2/css/sh_css_defs.h       |  1 +
 .../camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c   |  4 ++--
 13 files changed, 29 insertions(+), 12 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_api_version.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_api_version.h
index f2eac14..d016c34 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_api_version.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_api_version.h
@@ -34,7 +34,7 @@ The version string has four dot-separated numbers, read left to right:
 
 */
 
-#define CSS_API_VERSION_STRING	"2.1.20.8"
+#define CSS_API_VERSION_STRING	"2.1.20.9"
 
 /*
 Change log
@@ -432,6 +432,10 @@ v2.1.20.8, Temprarily disable memory debug features for SVOS.
 - Temporary commented out the additions to allow SKC testing till root cause found
 - Changed files ia_css_env.h and sh_css.c.
 
+v2.1.20.9, Enable ISP 2.7 naming
+- Add IA_CSS_PIPE_VERSION_2_7 to enum ia_css_pipe_version
+- Add #define SH_CSS_ISP_PIPE_VERSION_2_7 4
+
 */
 
 #endif /*__CSS_API_VERSION_H*/
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/hive_isp_css_include/mpmath.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/hive_isp_css_include/mpmath.h
index 48532ac..92fd528 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/hive_isp_css_include/mpmath.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/hive_isp_css_include/mpmath.h
@@ -58,6 +58,17 @@ typedef enum {
     N_mp_const_ID
 } mp_const_ID_t;
 
+/* _isValidMpudata is for internal use by mpmath and bbb's.
+ * isValidMpudata is for external use by functions on top.
+ */
+#ifndef ENABLE_VALID_MP_DATA_CHECK
+#define _isValidMpsdata(data,bitdepth) (1)
+#define _isValidMpudata(data,bitdepth) (1)
+#else
+#define _isValidMpsdata(data,bitdepth) isValidMpsdata(data,bitdepth)
+#define _isValidMpudata(data,bitdepth) isValidMpsdata(data,bitdepth)
+
+#endif
 STORAGE_CLASS_MPMATH_FUNC_H bool isValidMpsdata(
     const mpsdata_t             data,
     const bitdepth_t            bitdepth);
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
index 06e179d..c76460f 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_master_20150824_1500 */
+/* Release Version: irci_ecr-master_20150911_0724 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_host_data.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_host_data.h
index 31def93..43bb40a 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_host_data.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_host_data.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_master_20150824_1500 */
+/* Release Version: irci_ecr-master_20150911_0724 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_pipe_public.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_pipe_public.h
index 2ad1e86..03913bc 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_pipe_public.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_pipe_public.h
@@ -58,7 +58,8 @@ enum ia_css_pipe_mode {
 enum ia_css_pipe_version {
 	IA_CSS_PIPE_VERSION_1 = 1,		/**< ISP1.0 pipe */
 	IA_CSS_PIPE_VERSION_2_2 = 2,		/**< ISP2.2 pipe */
-	IA_CSS_PIPE_VERSION_2_6_1 = 3		/**< ISP2.6.1 pipe */
+	IA_CSS_PIPE_VERSION_2_6_1 = 3,		/**< ISP2.6.1 pipe */
+	IA_CSS_PIPE_VERSION_2_7 = 4		/**< ISP2.7 pipe */
 };
 
 /**
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
index 5c04e15..c14dcd6 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_master_20150824_1500 */
+/* Release Version: irci_ecr-master_20150911_0724 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
index 546040e..e293fe0 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
@@ -22,7 +22,7 @@ more details.
 #define __IA_CSS_VERSION_DATA_H
 
 
-#define CSS_VERSION_STRING "REL:20150824_35.1_2111; API:2.1.20.8; GIT:irci_20150823_1500_#ab52ca99b8af02136dec0975eb133a61098c869b; SDK:; USER:viedifw; "
+#define CSS_VERSION_STRING "REL:20150911_37.5_1652; API:2.1.20.9; GIT:irci___#ebf437d53a8951bb7ff6d13fdb7270dab393a92a; SDK:; USER:viedifw; "
 
 
 #endif
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr.host.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr.host.c
index c15ae1b..74d4a8a 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr.host.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr.host.c
@@ -1,4 +1,4 @@
-/* Release Version: irci_master_20150824_1500 */
+/* Release Version: irci_ecr-master_20150911_0724 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr.host.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr.host.h
index 0bda4df..ea8a99a 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr.host.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr.host.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_master_20150824_1500 */
+/* Release Version: irci_ecr-master_20150911_0724 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr_param.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr_param.h
index c32bf2a..5c3730c 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr_param.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr_param.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_master_20150824_1500 */
+/* Release Version: irci_ecr-master_20150911_0724 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr_types.h
index d9c9736..6873328 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr_types.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/hdr/ia_css_hdr_types.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_master_20150824_1500 */
+/* Release Version: irci_ecr-master_20150911_0724 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_defs.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_defs.h
index 386c1f1..e71e189 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_defs.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_defs.h
@@ -86,6 +86,7 @@ more details.
 #define SH_CSS_ISP_PIPE_VERSION_1	1
 #define SH_CSS_ISP_PIPE_VERSION_2_2	2
 #define SH_CSS_ISP_PIPE_VERSION_2_6_1	3
+#define SH_CSS_ISP_PIPE_VERSION_2_7	4
 
 /*--------------- sRGB Gamma -----------------
 CCM        : YCgCo[0,8191] -> RGB[0,4095]
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
index b9e87e0..43793a2 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
@@ -47,10 +47,10 @@ struct fw_param {
 /* Warning: same order as SH_CSS_BINARY_ID_* */
 static struct firmware_header *firmware_header;
 
-/* The string STR(irci_master_20150824_1500) is a place holder
+/* The string STR(irci_ecr-master_20150911_0724) is a place holder
  * which will be replaced with the actual RELEASE_VERSION
  * during package generation. Please do not modify  */
-static const char *release_version = STR(irci_master_20150824_1500);
+static const char *release_version = STR(irci_ecr-master_20150911_0724);
 
 #define MAX_FW_REL_VER_NAME	300
 static char FW_rel_ver_name[MAX_FW_REL_VER_NAME] = "---";
-- 
1.9.1

